ansari_db.py

```diff
@@ -426,19 +473,39 @@ def delete_thread(self, thread_id, user_id):
             if cur:
                 cur.close()
 
+    def delete_token(self, user_id, token, token_type):
+        try:
+            if token_type == "login":
+                db_table = "user_tokens"
+            elif token_type == "refresh":
+                db_table = "refresh_tokens"
+            else:
+                raise ValueError(f"Invalid token type {token_type}")
+            with self.conn.cursor() as cur:
+                delete_cmd = f"""DELETE FROM {db_table} WHERE user_id = %s AND token = %s;"""
+                cur.execute(delete_cmd, (user_id, token))
+                self.conn.commit()
+            return {"status": "success"}
+        except Exception as e:
+            logger.warning(f"Error is {e}")
+            return {"status": "failure", "error": str(e)}
+
     def logout(self, user_id):
         try:
-            cur = self.conn.cursor()
-            delete_cmd = """DELETE FROM user_tokens WHERE user_id = %s;"""
-            cur.execute(delete_cmd, (user_id,))
-            self.conn.commit()
+            # This will logout all tokens associated with the user.
+            # If the user has two active session on two different front-ends,
+            # logging out from one, will cause logout from all of them.
+            # Note that this is also related to the tokens table structure;
+            # there isn't a "token_id" column.
```

Comment:
Sorry about this. I talked to @ibrahim and he rightly pointed out this is a bug. This should take the token to be removed too.

Reply:
Done.